### PR TITLE
networkd: FOU netdev add support to configure peer port

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: man/systemd.netdev.xml [hunk 1]
```diff
@@ -1300,6 +1300,12 @@
           for delivery to the real destination. This option is mandatory.</para>
         </listitem>
         </varlistentry>
+      <varlistentry>
+        <term><varname>PeerPort=</varname></term>
+        <listitem>
+          <para>Specifies the peer port number. Defaults to unset.</para>
```
Comment (yuwata) [inline man/systemd.netdev.xml]: Please mention that this is mandatory when `Peer=` is set.

File: src/network/netdev/fou-tunnel.c [hunk 2]
```diff
@@ -47,6 +47,10 @@ static int netdev_fill_fou_tunnel_message(NetDev *netdev, sd_netlink_message **r
         if (r < 0)
                 return log_netdev_error_errno(netdev, r, "Could not append FOU_ATTR_PORT attribute: %m");
 
+        r = sd_netlink_message_append_u16(m, FOU_ATTR_PEER_PORT, htobe16(t->peer_port));
+        if (r < 0)
+                return log_netdev_error_errno(netdev, r, "Could not append FOU_ATTR_PEER_PORT attribute: %m");
```
Comment (yuwata) [inline src/network/netdev/fou-tunnel.c]: From the code in `netdev_fou_tunnel_verify()`, I guess that `peer_port` should be sent only when peer address is set. If that is true, then please do so.

File: src/network/netdev/fou-tunnel.c [hunk 3]
```diff
@@ -70,24 +74,28 @@ static int netdev_fill_fou_tunnel_message(NetDev *netdev, sd_netlink_message **r
...
+        if (t->local_family != AF_UNSPEC) {
+                if (t->local_family == AF_INET) {
+                        r = sd_netlink_message_append_in_addr(m, FOU_ATTR_LOCAL_V4, &t->local.in);
+                        if (r < 0)
+                                return log_netdev_error_errno(netdev, r, "Could not append FOU_ATTR_LOCAL_V4 attribute: %m");
+                } else {
+                        r = sd_netlink_message_append_in6_addr(m, FOU_ATTR_LOCAL_V6, &t->local.in6);
+                        if (r < 0)
+                                return log_netdev_error_errno(netdev, r, "Could not append FOU_ATTR_LOCAL_V6 attribute: %m");
+                }
```
Comment (yuwata) [inline src/network/netdev/fou-tunnel.c]: Not necessary to check `!= AF_UNSPEC`. Below can decrease indentation.
```
if (t->local_family == AF_INET) {
 ...
} else if if (t->local_family == AF_INET6) {
 ...
}
```

File: src/network/netdev/fou-tunnel.c [hunk 4]
```diff
@@ -70,24 +74,28 @@ static int netdev_fill_fou_tunnel_message(NetDev *netdev, sd_netlink_message **r
...
+        if (t->peer_family != AF_UNSPEC) {
+                if (t->peer_family == AF_INET) {
+                        r = sd_netlink_message_append_in_addr(m, FOU_ATTR_PEER_V4, &t->peer.in);
+                        if (r < 0)
+                                return log_netdev_error_errno(netdev, r, "Could not append FOU_ATTR_PEER_V4 attribute: %m");
+                } else {
+                        r = sd_netlink_message_append_in6_addr(m, FOU_ATTR_PEER_V6, &t->peer.in6);
+                        if (r < 0)
+                                return log_netdev_error_errno(netdev, r, "Could not append FOU_ATTR_PEER_V6 attribute: %m");
+                }
```
Comment (yuwata) [inline src/network/netdev/fou-tunnel.c]: Same here.

**Comment by ssahani** [comment]
updated thanks for the review @yuwata

File: man/systemd.netdev.xml [hunk 5]
```diff
@@ -1300,6 +1300,12 @@
           for delivery to the real destination. This option is mandatory.</para>
         </listitem>
         </varlistentry>
+      <varlistentry>
+        <term><varname>PeerPort=</varname></term>
+        <listitem>
+          <para>Specifies the peer port number. Defaults to unset. Note that when peer port is set <literal>Peer</literal> address is mandotory.</para>
```
Comment (yuwata) [inline man/systemd.netdev.xml]: Please append `=` to the setting name:
`<literal>Peer</literal>` -> `<literal>Peer=</literal>`.

File: man/systemd.netdev.xml [hunk 6]
```diff
@@ -1313,7 +1319,7 @@
       <varlistentry>
         <term><varname>Peer=</varname></term>
         <listitem>
-          <para>Configures peer IP address.</para>
+          <para>Configures peer IP address. Note that when peer address is set <literal>PeerPort</literal> is mandotory.</para>
```
Comment (yuwata) [inline man/systemd.netdev.xml]: Same here.

File: src/network/netdev/fou-tunnel.c [hunk 7]
```diff
@@ -74,20 +80,20 @@ static int netdev_fill_fou_tunnel_message(NetDev *netdev, sd_netlink_message **r
...
-                r = sd_netlink_message_append_in6_addr(m, FOU_ATTR_PEER_V6, &t->peer.in6);
-                if (r < 0)
-                        return log_netdev_error_errno(netdev, r, "Could not append FOU_ATTR_PEER_V6 attribute: %m");
+                        r = sd_netlink_message_append_in_addr(m, FOU_ATTR_PEER_V4, &t->peer.in);
+                        if (r < 0)
+                                return log_netdev_error_errno(netdev, r, "Could not append FOU_ATTR_PEER_V4 attribute: %m");
+        } else if (t->peer_family == AF_INET6){
+                        r = sd_netlink_message_append_in6_addr(m, FOU_ATTR_PEER_V6, &t->peer.in6);
+                        if (r < 0)
+                                return log_netdev_error_errno(netdev, r, "Could not append FOU_ATTR_PEER_V6 attribute: %m");
```
Comment (yuwata) [inline src/network/netdev/fou-tunnel.c]: Please fix indentation.

**Comment by ssahani** [comment]
Updated PTAL @yuwata thanks.

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**